marconi/src/Marconi/Indexers.hs

```diff
@@ -117,29 +117,31 @@ isInTargetTxOut targetAddresses (C.TxOut address _ _ _) = case address of
     _                                                -> False
 
 utxoWorker
-    :: (Utxo.UtxoIndex -> IO Utxo.UtxoIndex)    -- ^ CPS function used in the queryApi thread, needs to be non-blocking
-    -> Maybe TargetAddresses                    -- ^ Target addresses to filter for
-    -> Worker
-utxoWorker indexerCallback maybeTargetAddresses Coordinator{_barrier} ch path =
-    Utxo.open path (Utxo.Depth 2160) >>= indexerCallback >>= innerLoop
+  :: (Storable.StorableEvent Utxo.UtxoHandle -> IO [()])
+  -> (Utxo.UtxoIndexer -> IO Utxo.UtxoIndexer)  -- ^ CPS function used in the queryApi thread, needs to be non-blocking
```

Comment:
I think we can do better here. My dream would be to have `start....` return a `TVar UtxoIndex` so we can generically call whatever we want whenever we want. I am not sure my idea is better than a callback though. What do you think?

Reply:
the main reason for the `callback` is to provide access to the indexer's latest value,  to functions that run on a different thread, as is the case with mamba. I like your suggestion as it reduces the noise and simplifies the worker function type. However, I think it adds a slight complexity to the `RewindableIndex.Storable` cursor computation. That computation sill have to update the TVar in the `appendEvent` and `flushBufferEvent`. Small price to pay.  I like the idea, I can implement it in my Innovation - Marconi related work